packages/blade/src/components/Modal/_decisions/decisions.md

```diff
@@ -0,0 +1,152 @@
+
+Modal components are often used to present information, notifications, or requests that require immediate attention or confirmation from the user. 
+
+This document outlines the API of `Modal` component.
+
+<!-- TODO: Add this -->
+<!-- <img src="./modal-thumbnail.png" width="380" alt="Thumbnail of modal" /> -->
+
+- [Design](#design)
+- [Anatomy](#anatomy)
+- [Animation](#animation)
+- [API](#api)
+  - [`Modal` API](#modal-api)
+  - [`ModalBody` API](#modalbody-api)
+  - [`ModalHeader` API](#modalheader-api)
+  - [`ModalFooter` API](#modalfooter-api)
+- [Behaviors](#behaviors)
+  - [Modal Stacking](#modal-stacking)
+  - [Modal Sizes \& Responsiveness](#modal-sizes--responsiveness)
+  - [Modal Height](#modal-height)
+  - [Modal Usage on mWeb \& native apps](#modal-usage-on-mweb--native-apps)
+  - [Modal for React Native](#modal-for-react-native)
+- [Accessibility](#accessibility)
+- [Open questions](#open-questions)
+- [References](#references)
+
+## Design
+
+- [Figma - Modal](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=35860-607833&t=hEZsxB4yG6A3Gmib-0)
+
+## Anatomy
+<!-- TODO: Add a better screenshot -->
+<img src="./modal-anatomy.png" alt="Modal Anatomy" width="100%" />
+
+## Animation
+**Entry:**
+<img src="./modal-entry.gif" alt="Modal Entry Animation" width="100%" />
+
+
+**Exit:**
+<img src="./modal-exit.gif" alt="Modal Exit Animation" width="100%" />
+
+## API
+
+Sample usage:
+
+```jsx
+import { Modal, ModalHeader, ModalBody, ModalFooter, Button, Text } from '@razorpay/blade/components';
+
+const [isOpen, setIsOpen] = useState(false);
+
+<Modal isOpen={isOpen} onDismiss={() => setIsOpen(false)}>
+  <ModalHeader title="Modal Title" />
+  <ModalBody>
+    <Text>This is the Modal Body</Text>
+  </ModalBody>
+  <ModalFooter>
+    <Button variant='secondary'>Cancel</Button>
+    <Button variant='primary'>Save</Button>
+  </ModalFooter>
+</Modal>
+```
+
+### `Modal` API
+
+| Prop            | Type                       | Default             | Description                                                                                                                                                                | Required |
+| --------------- | -------------------------- | ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| isOpen          | `boolean`                  | `false`             | Toggles modal state                                                                                                                                                 |          |
+| size          | `small`, `medium`, `large`                  | `small`             | Size of the modal                                                                                                                                                 |          |
```

Comment:
can you elaborate the description for `size` prop? so we can understand how will size translate to `*width` properties